src/esp/physics/PhysicsManager.cpp

```diff
@@ -315,42 +315,42 @@ PhysicsManager::getArticulatedObjectWrapper() {
   return articulatedObjectManager_->createObject("ManagedArticulatedObject");
 }
 
-void PhysicsManager::removeObject(const int physObjectID,
+void PhysicsManager::removeObject(const int objectId,
                                   bool deleteObjectNode,
                                   bool deleteVisualNode) {
-  assertRigidIdValidity(physObjectID);
-  scene::SceneNode* objectNode = &existingObjects_.at(physObjectID)->node();
-  scene::SceneNode* visualNode = existingObjects_.at(physObjectID)->visualNode_;
-  std::string objName = existingObjects_.at(physObjectID)->getObjectName();
-  existingObjects_.erase(physObjectID);
-  deallocateObjectID(physObjectID);
+  assertRigidIdValidity(objectId);
+  scene::SceneNode* objectNode = &existingObjects_.at(objectId)->node();
+  scene::SceneNode* visualNode = existingObjects_.at(objectId)->visualNode_;
+  std::string objName = existingObjects_.at(objectId)->getObjectName();
+  existingObjects_.erase(objectId);
+  deallocateObjectID(objectId);
   if (deleteObjectNode) {
     delete objectNode;
   } else if (deleteVisualNode && visualNode) {
     delete visualNode;
   }
   // remove wrapper if one is present
   if (rigidObjectManager_->getObjectLibHasHandle(objName)) {
-    rigidObjectManager_->removeObjectByID(physObjectID);
+    rigidObjectManager_->removeObjectByID(objectId);
   }
 }  // PhysicsManager::removeObject
 
-void PhysicsManager::removeArticulatedObject(int physObjectID) {
-  CHECK(existingArticulatedObjects_.count(physObjectID));
+void PhysicsManager::removeArticulatedObject(int objectId) {
+  CHECK(existingArticulatedObjects_.count(objectId));
   scene::SceneNode* objectNode =
-      &existingArticulatedObjects_.at(physObjectID)->node();
+      &existingArticulatedObjects_.at(objectId)->node();
   for (auto linkObjId :
-       existingArticulatedObjects_.at(physObjectID)->objectIdToLinkId_) {
+       existingArticulatedObjects_.at(objectId)->objectIdToLinkId_) {
     deallocateObjectID(linkObjId.first);
   }
   std::string artObjName =
-      existingArticulatedObjects_.at(physObjectID)->getObjectName();
-  existingArticulatedObjects_.erase(physObjectID);
-  deallocateObjectID(physObjectID);
+      existingArticulatedObjects_.at(objectId)->getObjectName();
```

Comment:
Is C++ smart enough to cache these subsequent at calls? Or should we cache the return of the `existingArticulatedObjects_.at(objectId)` call?

Reply:
I'm really not sure if caching a local variable is a win here in general. Possibly once the number of objects in the world increases significantly this could be impactful. However, out of scope of this PR and possibly a larger refactor. Maybe motivates some benchmarking.